### ISSUE TITLE
Backup Endpoint responds with application/json

### ISSUE THREAD

**Issue by justinmorant** [rqlite/rqlite#333]
The endpoint /db/backup responds with an application/json Content-Type header which breaks HTTP clients that attempt to parse the response as JSON since it is plain text.

How to reproduce:
1) Start docker
```
docker pull rqlite/rqlite
docker run -p 4001:4001 -p4002:4002 rqlite/rqlite
```
2) Run commands in another terminal
```
curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d '[
    "CREATE TABLE foo (id integer not null primary key, name text)"
]'
curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d '[
    "INSERT INTO foo(name) VALUES(\"fiona\")"
]' 
curl localhost:4001/db/backup -o /dev/null -H "Accept: text/plain" -s -D -
```
3) Get result
```
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
X-Rqlite-Version: v4.0.0
Date: Sat, 29 Jul 2017 17:09:17 GMT
Transfer-Encoding: chunked
```

**Comment by justinmorant** [comment]
@otoolep for rqlite-js I am blocked on writing the backup and restore endpoints until we can fix this issue. I would have to hack around the client so it does not attempt to parse the body into Javascript objects when 'Content-Type: application/json' is present.  Let me know if this will take a while to fix since I can make the client work, just want to avoid the change.

**Comment by otoolep** [comment]
Thanks @justinmorant -- real issue, but easy to fix. The main fix is in the PR above. 

What do you suggest as a Content-type for the backup response? I believe `application/octet-stream` is most sensible: https://www.iana.org/assignments/media-types/application/octet-stream

**Comment by otoolep** [comment]
OK, `octet-stream` looks like the most meaningful MIME type. I'll merge @justinmorant if this works for you.

**Comment by otoolep** [comment]
Decided to merge since the header type makes most sense to me. Easy to change if necessary.

**Comment by justinmorant** [comment]
@otoolep thanks that is fine with me, as long as it does not have a reference to JSON in the header most libraries won't try and parse as such. The `application/octet-stream` content type is pretty abstract with regards to implementation so we should not run into any issues.

**Comment by otoolep** [comment]
Fixed in 4.0.1.
